### PR TITLE
ci(scripts): allow dry release

### PR DESCRIPTION
**Pull request by JeremyDec**
ref: MANAGER-9039
Signed-off-by: JeremyDec <jeremy.de-cesare@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-9039
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally (Script)
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: scripts/common/repository.js [hunk 1]
```diff
@@ -70,9 +70,11 @@ class MonoRepository {
...
       )
       .then((v) =>
-        execa
-          .command('git push origin master --tags', { shell: true })
-          .then(() => v),
+        !dryRelease
+          ? execa
+              .command('git push origin master --tags', { shell: true })
+              .then(() => v)
+          : Promise.resolve(v),
```
Comment (frenautvh) [inline scripts/common/repository.js:77]: nitpicking for readability

```suggestion
        dryRelease
          ? Promise.resolve(v)
          : execa
              .command('git push origin master --tags', { shell: true })
              .then(() => v),
```

File: scripts/common/repository.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ class MonoRepository {
       });
   }
 
-  static release(version, repos) {
+  static release(version, repos, dryRelease) {
```
Comment (marie-j) [inline scripts/common/repository.js:58]: suggestion: 
```suggestion
  static release(version, repos, dryRelease = false) {
```
(even though with JS `!dryrelease` will result with the same)

**Review by cbourgois — DISMISSED** [review]

**Review by marie-j — APPROVED** [review]

**Comment by JeremyDec** [comment]
commit prefix + target change since last approvals

**Review by marie-j — APPROVED** [review]

File: scripts/common/repository.js [hunk 3]
```diff
@@ -186,7 +188,7 @@ class Repository {
   }
 
   getPackageJson() {
-    return require(this.getPackageJsonPath());
+    return require(this.getPackageJsonPath()); // eslint-disable-line
```
Comment (JeremyDec) [inline scripts/common/repository.js:191]: @antleblanc the disabling eslint comment has been moved here, using global `eslint-disable-line` as the super-linter action doesn't recognize rule `import/no-dynamic-require`.

(Error is: Definition for rule 'import/no-dynamic-require' was not found)

**Review by antleblanc — APPROVED** [review]

**Pull request merged**